hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BPOfferService.java

```diff
@@ -831,6 +835,11 @@ public Object call() throws Exception {
       if (dn.blockScanner != null) {
         dn.blockScanner.addBlockPool(getBlockPoolId());
       }
+
+      if (brFailed){
+        Thread.sleep(2000); //sleep before retrying to send the block report
```

Comment:
Without this failed block report will be retired every second. I think it is too quick. with 2 sec added delay block report will be retried after total of three seconds delay. The sleep happens after the cache reports. I do not see any issue here.

Reply:
I'm not a fan of having delay accumulating on each other because it makes it difficult to have a good idea of the actual delay in the end and it will delay the next cache report by 2s instead of 1s.
But I it should be ok like that.